handlers/validator.go

```diff
@@ -1811,10 +1827,15 @@ func ValidatorSync(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	//remove all future sync periods
 	latestEpoch := services.LatestEpoch()
-	for syncPeriods[0].EndEpoch > services.LatestEpoch() {
-		syncPeriods = syncPeriods[1:]
+
+	//remove scheduled committees
+	for _, syncPeriod := range syncPeriods {
+		if syncPeriod.EndEpoch > latestEpoch {
+			syncPeriods = syncPeriods[1:]
+		} else {
+			break
+		}
 	}
```

Comment:
```
for i, syncPeriod := range syncPeriods {
	if syncPeriod.EndEpoch <= latestEpoch {
		syncPeriods = syncPeriods[i:]
		break
	}
}
```

Reply:
Maybe faster?

Reply:
Is it guaranteed that no more than 1 sync committee is scheduled? Otherwise you could also just do
`if syncPeriods[0].EndEpoch > latestEpoch {
	syncPeriods = syncPeriods[1:]
}`

Reply:
No idea, but based on the original code, I guess no guarantee at all :S

Reply:
Let us keep in mind that the same exact issue appears twice: Once here and once in line 634.

Reply:
In order to implement the changes suggested in [my comment](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1814#issuecomment-1436570717) I will keep the code as is and not implement your suggested changes.

Reply:
Nvm, did not read through @recy21 's suggestion correctly, implemented in [new commit](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1814/commits/ecedce684e794aa20496267997b91decca420ffd)